kernel/syscall.c

```diff
@@ -196,17 +196,55 @@ static bool _trace_syscall(long n)
     // Check if syscall tracing is enabled.
     if (__myst_kernel_args.strace_config.trace_syscalls)
     {
+        // -1 represents strace syscall filter is not set
+        short filter_syscall = -1;
+
         if (__myst_kernel_args.strace_config.filter)
         {
-            // If filtering is enabled, trace only this syscall has been
+            // If filtering is enabled, trace only this syscall if it has been
             // specified in the filter.
-            return __myst_kernel_args.strace_config.trace[n];
+            filter_syscall = __myst_kernel_args.strace_config.trace[n];
```

Comment:
Can we declare `filter_syscall ` here

`bool filter_syscall = __myst_kernel_args.strace_config.trace[n];`

Reply:
removed the variable and changed the logic